packages/vsce/src/commands/showLogsCommand.ts

```diff
@@ -0,0 +1,122 @@
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { commands, TreeView, window } from "vscode";
+import { CICSRegionTree } from "../trees/CICSRegionTree";
+import { findSelectedNodes } from "../utils/commandUtils";
+import { ProfileManagement } from "../utils/profileManagement";
+import { Gui } from "@zowe/zowe-explorer-api";
+import { getResource, ICMCIApiResponse } from "@zowe/cics-for-zowe-sdk";
+import * as vscode from "vscode";
+import { IProfileLoaded } from "@zowe/imperative";
+
+export async function findRelatedZosProfiles(cicsProfile: IProfileLoaded, zosProfiles: IProfileLoaded[]): Promise<IProfileLoaded> {
+  const baseForCicsProfile = await ProfileManagement.getProfilesCache().fetchBaseProfile(cicsProfile.name);
+
+  // sort profiles with zosmf ones first to make zosmf the default
+  // also filter so we only automatically pick z/os connections that have credentials associated
+  zosProfiles = zosProfiles.sort((prof) => (prof.profile.type === "zosmf" ? -1 : 1)).filter((prof) => prof.profile.user);
+
+  // filter out profiles that are not in the same base as the cics profile
+  const matchingProfiles: IProfileLoaded[] = [];
+  if (baseForCicsProfile) {
+    for (const profile of zosProfiles) {
+      if (baseForCicsProfile && baseForCicsProfile?.name === (await ProfileManagement.getProfilesCache().fetchBaseProfile(profile.name))?.name) {
+        matchingProfiles.push(profile);
+      }
+    }
+  }
+
+  if (matchingProfiles.length > 0) {
+    return matchingProfiles[0];
+  }
+
+  // we couldn't find anything within a profile group
+  // filter down to just profiles that have the same hostname as our cics connection
+  const cicsHostProfiles = zosProfiles.filter((profile) => cicsProfile.profile.host === profile.profile.host);
+
+  return cicsHostProfiles[0];
+}
+
+function promptUserForProfile(zosProfiles: IProfileLoaded[]): Thenable<string> {
+  const profileNames = zosProfiles.map((value) => {
+    return value.name;
+  });
+
+  if (profileNames.length > 0) {
+    const quickPickOptions: vscode.QuickPickOptions = {
+      placeHolder: vscode.l10n.t("Select a profile to access the logs"),
+      ignoreFocusOut: true,
+      canPickMany: false,
+    };
+    return Gui.showQuickPick(profileNames, quickPickOptions);
+  } else {
+    return null;
+  }
+}
+
+export async function getJobIdForRegion(selectedRegion: CICSRegionTree): Promise<string> {
+  // when we have a CICSRGN table we have jobid, but not when we have a
+  // MAS table. we get either CICSRGN or MAS depending on where we are in the
+  // tree. request CICSRGN if jobid isn't available.
+  let jobid: string = selectedRegion.region.jobid;
+  if (!jobid) {
+    let response: ICMCIApiResponse;
+    try {
+      response = await getResource(selectedRegion.parentSession.getSession(), {
+        name: "CICSRegion",
```

Comment:
We have a constant for this - and if we don't, we should!

Reply:
good shout - will be in the next push.